packages/core/src/stepper/stepper-root.tsx

```diff
@@ -0,0 +1,95 @@
+import { Tabs } from "../tabs";
+import { mergeDefaultProps } from "@kobalte/utils";
+import {
+	type JSX,
+	type ValidComponent,
+	createUniqueId,
+	splitProps,
+	createSignal,
+} from "solid-js";
+
+import { type ElementOf, type PolymorphicProps } from "../polymorphic";
+import { StepperContext, type StepperContextValue } from "./stepper-context";
+
+export interface StepperRootOptions {
+	/** The controlled step of the stepper. */
+	step?: number;
+
+	/** The default step when initially rendered. Useful when you do not need to control the step state. */
+	defaultStep?: number;
+
+	/** Event handler called when the step changes. */
+	onStepChange?: (step: number) => void;
+
+	/** Whether the stepper is disabled. */
+	disabled?: boolean;
+
+  /** The maximum number of steps in the stepper. */
+  maxSteps: number;
```

Comment:
are you suggesting, to handle this maxSteps count internally instead of having this as `prop`? or suggesting it to change it's `name`

Reply:
In my opinion, it would be more effective to manage the `maxSteps` count internally rather than exposing it as a prop. This approach will minimize the potential for human errors and simplify the component overall. You might consider adding `context.register` in the `onMount` and `context.unregister` in the `onCleanup` method in the `Stepper.Content` and maybe `Stepper.CompletedContent`. These functions can be used to increment or decrement the `stepsCount` or `length` prop accordingly.